extensions/csrc/kernel/cuda/utils/vec_copy.h

```diff
@@ -33,9 +32,33 @@ __device__ __inline__ void copy_zero_vector(T *dst) {
   *(reinterpret_cast<VT *>(dst)) = funcs::CastFunctor<float, VT>()(0.0f);
 }
 
+template <typename SrcT, typename DstT, int vec_size>
+__device__ __inline__ void copy(const SrcT *src, DstT *dst) {
+  using SrcVT = typename common::VecTypeTrait<SrcT, vec_size>::Type;
+  using DstVT = typename common::VecTypeTrait<DstT, vec_size>::Type;
+  // Note(LiuYang): Here static_cast can't be used for cast between two pointer
+  *(reinterpret_cast<DstVT *>(dst)) = funcs::CastFunctor<SrcVT, DstVT>()(
```

Comment:
Wonder if `static_cast` here in `CastFunctor` was accidentally added, or it (only) aims at compile-time check?

Reply:
Next PR I will delete this note